# Actividad 2

1. Importar librerias

In [28]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

2. Leer archivo

In [29]:
df = pd.read_csv("/content/titanic.csv")

3. Observar primeras lineas

In [30]:
df.head()

PassengerId  Survived  Pclass  \
0            1         0       3   
1            2         1       1   
2            3         1       3   
3            4         1       1   
4            5         0       3   

                                                Name     Sex   Age  SibSp  \
0                            Braund, Mr. Owen Harris    male  22.0      1   
1  Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                             Heikkinen, Miss. Laina  female  26.0      0   
3       Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
4                           Allen, Mr. William Henry    male  35.0      0   

   Parch            Ticket     Fare Cabin Embarked  
0      0         A/5 21171   7.2500   NaN        S  
1      0          PC 17599  71.2833   C85        C  
2      0  STON/O2. 3101282   7.9250   NaN        S  
3      0            113803  53.1000  C123        S  
4      0            373450   8.0500   NaN        S

4. Obtener diferentes combinaciones de columnas y datos

In [31]:
print(df[["Survived","Pclass","Sex"]])

     Survived  Pclass     Sex
0           0       3    male
1           1       1  female
2           1       3  female
3           1       1  female
4           0       3    male
..        ...     ...     ...
886         0       2    male
887         1       1  female
888         0       3  female
889         1       1    male
890         0       3    male

[891 rows x 3 columns]


In [32]:
print(df.iloc[0:10, [1,3,5]])

   Survived                                               Name   Age
0         0                            Braund, Mr. Owen Harris  22.0
1         1  Cumings, Mrs. John Bradley (Florence Briggs Th...  38.0
2         1                             Heikkinen, Miss. Laina  26.0
3         1       Futrelle, Mrs. Jacques Heath (Lily May Peel)  35.0
4         0                           Allen, Mr. William Henry  35.0
5         0                                   Moran, Mr. James   NaN
6         0                            McCarthy, Mr. Timothy J  54.0
7         0                     Palsson, Master. Gosta Leonard   2.0
8         1  Johnson, Mrs. Oscar W (Elisabeth Vilhelmina Berg)  27.0
9         1                Nasser, Mrs. Nicholas (Adele Achem)  14.0


5. Revisar los datos obtenidos por la funcion describe

In [33]:
df.describe()

PassengerId    Survived      Pclass         Age       SibSp  \
count   891.000000  891.000000  891.000000  714.000000  891.000000   
mean    446.000000    0.383838    2.308642   29.699118    0.523008   
std     257.353842    0.486592    0.836071   14.526497    1.102743   
min       1.000000    0.000000    1.000000    0.420000    0.000000   
25%     223.500000    0.000000    2.000000   20.125000    0.000000   
50%     446.000000    0.000000    3.000000   28.000000    0.000000   
75%     668.500000    1.000000    3.000000   38.000000    1.000000   
max     891.000000    1.000000    3.000000   80.000000    8.000000   

            Parch        Fare  
count  891.000000  891.000000  
mean     0.381594   32.204208  
std      0.806057   49.693429  
min      0.000000    0.000000  
25%      0.000000    7.910400  
50%      0.000000   14.454200  
75%      0.000000   31.000000  
max      6.000000  512.329200

Al revisar la estructura de nuestro dataset, podemos ver una mezcla de distintos tipos de datos. Tenemos variables puramente numéricas que nos servirán para cálculos, como Age y Fare, que representan la edad y la tarifa pagada. También contamos con SibSp y Parch, que son números discretos que nos indican la cantidad de familiares que viajaban con cada pasajero. Por otro lado, aunque la columna Pclass está registrada como un número entero, en realidad funciona como una variable ordinal, ya que la primera, segunda y tercera clase representan una jerarquía clara. El resto de las columnas son en su mayoría categóricas, como Sex o Embarked, o identificadores únicos como Name y Ticket

# Actividad 3

Cantidad de filas y columnas:

In [34]:
# Numero de filas y columnas del dataset
print(f"Filas del data set: {df.shape[0]}")
print(f"Columnas del data set: {df.shape[1]}")

Filas del data set: 891
Columnas del data set: 12


Variables y tipos de datos:

In [35]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 891 entries, 0 to 890
Data columns (total 12 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   PassengerId  891 non-null    int64  
 1   Survived     891 non-null    int64  
 2   Pclass       891 non-null    int64  
 3   Name         891 non-null    object 
 4   Sex          891 non-null    object 
 5   Age          714 non-null    float64
 6   SibSp        891 non-null    int64  
 7   Parch        891 non-null    int64  
 8   Ticket       891 non-null    object 
 9   Fare         891 non-null    float64
 10  Cabin        204 non-null    object 
 11  Embarked     889 non-null    object 
dtypes: float64(2), int64(5), object(5)
memory usage: 83.7+ KB


Al revisar la estructura de nuestro dataset, podemos ver una mezcla de distintos tipos de datos. Tenemos variables puramente numéricas que nos servirán para cálculos, como Age y Fare, que representan la edad y la tarifa pagada. También contamos con SibSp y Parch, que son números discretos que nos indican la cantidad de familiares que viajaban con cada pasajero. Por otro lado, aunque la columna Pclass está registrada como un número entero, en realidad funciona como una variable ordinal, ya que la primera, segunda y tercera clase representan una jerarquía clara. El resto de las columnas son en su mayoría categóricas, como Sex o Embarked, o identificadores únicos como Name y Ticket.

Para este paso vamos a checar las variables numericas, ya que las ordinales no tiene mucho sentido sacar las estadisticas descriptivas, como lo son el sexo, embarked y los identificadores unicos

In [36]:
## Seleccionamos las columnas que son numericas para hacer el analisis de sus medias, medianas y modas
col_num = ['Age', "Fare","SibSp","Parch"]

print(f"Media del dataset:\n {df[col_num].mean()}")
print(f"Mediana del dataset:\n {df[col_num].median()}")
print(f"Desviacion Estandar del dataset:\n {df[col_num].std()}")

Media del dataset:
 Age      29.699118
Fare     32.204208
SibSp     0.523008
Parch     0.381594
dtype: float64
Mediana del dataset:
 Age      28.0000
Fare     14.4542
SibSp     0.0000
Parch     0.0000
dtype: float64
Desviacion Estandar del dataset:
 Age      14.526497
Fare     49.693429
SibSp     1.102743
Parch     0.806057
dtype: float64


Al revisar los datos, podemos ver que cada variable se comporta de una forma diferente. Para la Edad, notamos que la distribución es bastante simétrica. Esto lo podemos ver porque su media (29.7) y su mediana (28.0) son valores muy parecidos. La desviación estándar de 14.5 nos indica también que había una gran variedad en las edades de los pasajeros. Por otro lado, la Tarifa es todo lo contrario, es muy asimétrica. Su media de 32.2 es mucho mayor que la mediana de 14.45, lo que nos muestra un claro sesgo a la derecha. Esto suele pasar cuando unos pocos valores muy altos "jalan" el promedio hacia arriba, y la alta desviación estándar (49.7) lo confirma.

Con las variables que cuentan familiares, SibSp y Parch, pasa algo interesante. Lo más importante es que en ambas la mediana es 0. Esto significa que la mayoría de los pasajeros no viajaban con hermanos, cónyuges, padres o hijos. Al tener tantos valores en cero, estas distribuciones también están muy sesgadas. En resumen, mientras la edad se distribuye de una manera más o menos normal, las variables de la tarifa y los familiares muestran datos muy concentrados en un extremo.